### PR TITLE
Avoid redefining Frame._get_columns_by_label in subclasses

### PR DESCRIPTION
**Pull request by mroeschke**
## Description
`Frame._get_columns_by_label` was redefined in `Series` and `DataFrame` to handle some special edge cases in `DataFrame.__getitem__` and empty `Series`

By making `_from_data_like_self` more consistent in preserving external properties and moving special casing, we can only define `Frame._get_columns_by_label` once

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/core/frame.py [hunk 1]
```diff
@@ -1434,22 +1442,18 @@ def _get_sorted_inds(
...
-            *(
-                self
-                if by is None
-                else self._get_columns_by_label(list(by), downcast=False)
-            )._columns
-        ]
+        if by is None:
+            to_sort = self._columns
+        else:
+            to_sort = self._get_columns_by_label(list(by))._columns
```
Comment (charlesbluca) [inline python/cudf/cudf/core/frame.py:1448]: Outside of dropping of `downcast`, is there any reason for the restructure here? Are there cases where we would expect the unpacking here not to work?

**Review by charlesbluca — APPROVED** [review]
Thanks @mroeschke! One question but overall LGTM

File: python/cudf/cudf/core/frame.py (hunk 1 above)
Comment (mroeschke) [reply python/cudf/cudf/core/frame.py:1448]: This was mostly a micro-optimization to avoid the unnecessary unpacking of the `if by is None` case

**Pull request merged**